test/storage/b_plus_tree_insert_test.cpp

```diff
@@ -39,6 +39,49 @@ TEST(BPlusTreeTests, DISABLED_InsertTest1) {
   auto header_page = bpm->NewPage(&page_id);
   (void)header_page;
 
+  int64_t key = 42;
+  int64_t value = key & 0xFFFFFFFF;
+  rid.Set(static_cast<int32_t>(key >> 32), value);
+  index_key.SetFromInteger(key);
+  tree.Insert(index_key, rid, transaction);
+
+  auto root_page_id = tree.GetRootPageId();
+  auto root_page = reinterpret_cast<BPlusTreePage *>(bpm->FetchPage(root_page_id)->GetData());
+  ASSERT_NE(root_page, nullptr);
+  ASSERT_TRUE(root_page->IsLeafPage());
+
+  auto root_as_leaf = reinterpret_cast<BPlusTreeLeafPage<GenericKey<8>, RID, GenericComparator<8>> *>(root_page);
+  ASSERT_EQ(root_as_leaf->GetSize(), 1);
+  ASSERT_EQ(comparator(root_as_leaf->KeyAt(0), index_key), 0);
```

Comment:
hmm, so it's zero 🤤

Reply:
The actual value used will be the RID which is constructed using CONCAT(key>>32, value) I believe, so it should not be zero. In this case, this is using the comparator to compare the key, not the value. This should be 42 I believe.